include/mbedtls/ssl.h

```diff
@@ -2424,6 +2467,32 @@ typedef int mbedtls_ssl_export_keys_ext_t( void *p_expkey,
                                            const unsigned char client_random[32],
                                            const unsigned char server_random[32],
                                            mbedtls_tls_prf_types tls_prf_type );
+#endif /* MBEDTLS_SSL_PROTO_TLS1 || MBEDTLS_SSL_PROTO_TLS1_1 || \
+          MBEDTLS_SSL_PROTO_TLS1_2 */
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+/**
+ * \brief           Callback type: Export TLS 1.3 secrets (client early traffic
+ *                  secret, client handshake traffic secret, server handshake
+ *                  traffic secret, client application 0 traffic secret, server
+ *                  application 0 traffic secret, exporter master secret)
+ *
+ * \note            This is required for certain uses of TLS, e.g. EAP-TLS
+ *                  (RFC 5216) and Thread. The key pointers are ephemeral and
+ *                  therefore must not be stored. The master secret and keys
+ *                  should not be used directly except as an input to a key
+ *                  derivation function.
+ *
+ * \param p_expkey  Context for the callback
+ * \param secret    Pointer to secret
+ * \param len       Secret length
+ *
+ * \return          0 if successful, or
+ *                  a specific MBEDTLS_ERR_XXX code.
+ */
+typedef int mbedtls_ssl_export_secret_t( void *p_expkey,
```

Comment:
Suggestion: Instead of having one cb per secret, can we use a single callback which is called multiple times, once for each piece of data?

Reply:
Something like: `mbedtls_ssl_export_secret_t( void *p_exp, mbedtls_ssl_tls1_3_secret_type type, const unsigned char *secret, size_t secret_len );`.